### ISSUE TITLE
Accessibility: repeated label on BottomNavigationBarItem

### ISSUE THREAD

**Issue by soyangel** [flutter/flutter#71543]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Modify main.dart like this:
``` dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _currentIndex = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      bottomNavigationBar: BottomAppBar(
        child: BottomNavigationBar(
          currentIndex: _currentIndex,
          onTap: (i) {
            setState(() {
              _currentIndex = i;
            });
          },
          items: [
            BottomNavigationBarItem(
              icon: Icon(Icons.bluetooth_searching),
              label: "Dog",
            ),
            BottomNavigationBarItem(
              icon: Icon(Icons.star_border),
              label: "Cat",
            )
          ],
        ),
      ),
    );
  }
}
```
3. Enable TalkBack on Android and select one of the bottom navigation tabs

(Should be similar on iOS VoiceOver)

**Expected results:** 

TalkBack should say something like "Selected dog tab one of two"

**Actual results:** 

Instead it says "Selected dog dog tab one of two"

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel stable, 1.22.4, on Linux, locale es_ES.UTF-8)
    • Flutter version 1.22.4 at /home/angel/flutter/soft/flutter
    • Framework revision 1aafb3a8b9 (hace 3 semanas), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /home/angel/android/soft/sdk/
    • Platform android-30, build-tools 30.0.0
    • ANDROID_HOME = /home/angel/android/soft/sdk/
    • ANDROID_SDK_ROOT = /home/angel/android/soft/sdk/
    • Java binary at: /home/angel/android/soft/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at /home/angel/android/soft/android-studio
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.51.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.16.0

[✓] Connected device (1 available)
    • Pixel 3a (mobile) • 93JAY0BMB2 • android-arm64 • Android 11 (API 30)

• No issues found!
```

</details>

You can fix this problem editing bottom_navigation_bar.dart and replacing:

``` dart
    if (item.label != null) {
      result = Tooltip(
        message: item.label,
        preferBelow: false,
        verticalOffset: selectedIconSize + selectedFontSize,
        child: result,
      );
    }
```

with this:

``` dart
    if (item.label != null) {
      result = Tooltip(
        message: item.label,
        preferBelow: false,
        verticalOffset: selectedIconSize + selectedFontSize,
        child: ExcludeSemantics(child: result), // <-- FIX HERE
      );
    }
```

**Comment by TahaTesser** [comment]
Hi @soyangel 
Thanks for filing the issue, I can reproduce the issue, iOS & Android and the fix does work as well

<details> 
<summary>code sample</summary> 

```bash
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _currentIndex = 0;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      bottomNavigationBar: BottomAppBar(
        child: BottomNavigationBar(
          currentIndex: _currentIndex,
          onTap: (i) {
            setState(() {
              _currentIndex = i;
            });
          },
          items: [
            BottomNavigationBarItem(
              icon: Icon(Icons.bluetooth_searching),
              label: "Dog",
            ),
            BottomNavigationBarItem(
              icon: Icon(Icons.star_border),
              label: "Cat",
            )
          ],
        ),
      ),
    );
  }
}

```

</details>


<details> 
<summary>multiple flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.4, on macOS 11.0.1 20B29 darwin-x64, locale
    en-GB)
    • Flutter version 1.22.4 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 1aafb3a8b9 (3 weeks ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (2 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64 • Android
      10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios           • iOS
      14.2

! Doctor found issues in 1 category.
```

```bash
[✓] Flutter (Channel master, 1.24.0-8.0.pre.406, on macOS 11.0.1 20B29
    darwin-x64, locale en-GB)
    • Flutter version 1.24.0-8.0.pre.406 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision 60bdcf55fe (6 hours ago), 2020-12-01 21:24:21 -0800
    • Engine revision 20caf54969
    • Dart version 2.12.0 (build 2.12.0-76.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.2)
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.51.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.16.0

[✓] Connected device (5 available)
    • RMX2001 (mobile)     • EUYTFEUSQSRGDA6D          • android-arm64  •
      Android 10 (API 29)
    • Taha’s iPad (mobile) • 00008020-000255113EE8402E • ios            • iOS
      14.2
    • macOS (desktop)      • macos                     • darwin-x64     • macOS
      11.0.1 20B29 darwin-x64
    • Web Server (web)     • web-server                • web-javascript •
      Flutter Tools
    • Chrome (web)         • chrome                    • web-javascript • Google
      Chrome 87.0.4280.67

• No issues found!
```

</details>

**Comment by yash1200** [comment]
I would like to land a PR for this issue.

**Comment by rydmike** [comment]
Wow you are on fire @yash1200 👍🏻  
This issue does go well together with https://github.com/flutter/flutter/issues/71049, kind of related.
